waspc/cli/Command/Telemetry/Project.hs

```diff
@@ -89,17 +92,31 @@ initialCache = ProjectTelemetryCache { _lastCheckIn = Nothing, _lastCheckInBuild
 
 -- * Project telemetry cache file.
 
-readOrCreateProjectTelemetryFile :: Path Abs (Dir TelemetryCacheDir) -> ProjectHash -> IO ProjectTelemetryCache
-readOrCreateProjectTelemetryFile telemetryCacheDirPath projectHash = do
-    fileExists <- SD.doesFileExist filePathFP
-    maybeCache <- if fileExists then readCacheFile else return Nothing
-    case maybeCache of
-        Just cache -> return cache
-        Nothing -> writeProjectTelemetryFile telemetryCacheDirPath projectHash initialCache >> return initialCache
+convertMaybeTimeToString :: Maybe T.UTCTime -> String
+convertMaybeTimeToString = maybe "nil" show
+
+showLastSentTelemetryDateTime :: Maybe ProjectTelemetryCache -> String
+showLastSentTelemetryDateTime maybeCache = case maybeCache of
+    Just cache -> "Telemetry data was last sent at: " <> convertMaybeTimeToString (_lastCheckIn cache) <>
+                  "\nTelemetry data was last upon using `build` was at: " <> convertMaybeTimeToString (_lastCheckInBuild cache)
+    Nothing    -> "Failed to find/read the telemetry file to display sufficient information"
+
+readProjectTelemetryFile :: Path Abs (Dir TelemetryCacheDir) -> ProjectHash -> Maybe ProjectTelemetryCache
```

Comment:
I am very confused about the type signature or this function -> how can it return `Maybe ProjectTelemetryCache` when you have IO calls inside? This shouldn't compile.

Reply:
Actually you can saying it failing to compile here: https://github.com/wasp-lang/wasp/pull/195/checks?check_run_id=1988613785 .